NDTensors/src/dense/fill.jl

```diff
@@ -21,19 +21,26 @@ function generic_randn(StoreT::Type{<:Dense}, dim::Integer=0)
   return generic_randn(default_storagetype(), dim)
 end
 
-function generic_zeros(
-  StoreT::Type{<:Dense{ElT,DataT}}, dim::Integer=0
-) where {DataT<:AbstractArray,ElT}
-  @assert ElT == eltype(DataT)
-  data = generic_zeros(DataT, dim)
-  StoreT = set_datatype(StoreT, typeof(data))
-  return StoreT(data)
+function generic_zeros(StoreT::Type{<:Dense}, dim::Integer)
+  return generic_zeros(StoreT, (dim,))
 end
 
-function generic_zeros(StoreT::Type{<:Dense{ElT}}, dim::Integer=0) where {ElT}
-  return generic_zeros(default_storagetype(ElT), dim)
+using .TypeParameterAccessors:
+  default_type_parameter,
+  parenttype,
+  set_eltype,
+  specify_default_type_parameters,
+  type_parameter
+function generic_zeros(StoreT::Type{<:Dense}, dims::Tuple{Integer})
+  StoreT = specify_default_type_parameters(StoreT)
+  DataT = specify_type_parameter(type_parameter(StoreT, parenttype), eltype, eltype(StoreT))
+  @assert eltype(StoreT) == eltype(DataT)
+
+  data = generic_zeros(DataT, dim(dims))
+  StoreT = set_datatype(StoreT, typeof(data))
+  return StoreT(data)
 end
 
-function generic_zeros(StoreT::Type{<:Dense}, dim::Integer=0)
-  return generic_zeros(default_storagetype(), dim)
+function generic_zeros(::Type{<:Dense}, ::Tuple{Integer,Integer,Vararg{Integer}})
+  return error("Can't make a multidimensional `Dense` object.")
```

Comment:
Instead of this definition just for catching issues with `Dense`, I think we should make a more systematic fix to the generic `generic_zeros` and `generic_randn` definitions. I think they are too general and should be constrained such that `ndims(arraytype)` == length(dims)`.

Reply:
It is not a constraint that `generic_zeros` only take `dim::Integer` 
```
julia> NDTensors.generic_zeros(Array, (2,3))
2×3 Matrix{Float64}:
 0.0  0.0  0.0
 0.0  0.0  0.0
 ``` 
 The constraint is that `Dense` only accepts Vectors.
This issue does exist for `generic_randn(::Type{<:AbstractArray})` so I fixed that.

Reply:
Right, but what I mean is that right now the current version allows `NDTensors.generic_zeros(Vector, (2,3))` which I don't think it should. More generally, I don't think it makes sense to allow inputting an array type where the dimensions are specified and where `dims` doesn't match that specified number of dimensions.

Reply:
I agree that if the dimensions aren't specified, like in your example `generic_zeros(Array, (2, 3))`, then any length `dims` should be allowed.

Reply:
Okay I was able to update AbstractArrays `generic_zeros` and `generic_randn` to verify that the length of dims = ndims(Arraytype) but it does require `TypeParameterAccessors.position(::Type, ::typeof(ndims))` to be defined